tutorial/beginner/t11_debugging.ipynb

```diff
@@ -0,0 +1,854 @@
+      "(3,)\n",
+      "TensorShape([3, 10])\n",
+      "(3,)\n",
+      "TensorShape([3, 10])\n",
+      "(3,)\n"
+     ]
+    }
+   ],
+   "source": [
+    "estimator = fe.Estimator(pipeline=pipeline,\n",
+    "                         network=network,\n",
+    "                         epochs=2,\n",
+    "                         train_steps_per_epoch=2, \n",
+    "                         eval_steps_per_epoch=1,\n",
+    "                         log_steps=None)\n",
+    "estimator.fit(eager=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b9ff05f",
+   "metadata": {},
+   "source": [
+    "### Verify and Debug the Network Results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3ae6f3e7",
+   "metadata": {},
+   "source": [
+    "Now let's look at how to verify the output of the network using `network.transform`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "35fbf93f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "network = fe.Network(ops=[ModelOp(model=model, inputs=\"x_out\", outputs=\"y_pred\"), #default mode=None\n",
+    "                          CrossEntropy(inputs=(\"y_pred\", \"y\"), outputs=\"ce\", mode=\"!infer\"),\n",
+    "                          UpdateOp(model=model, loss_name=\"ce\", mode=\"train\")])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d8081eb",
+   "metadata": {},
+   "source": [
+    "We will take the output of the pipeline and feed it into the network to verify that the network is giving us the output as was expected"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "736f90a7",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Labels:  tf.Tensor([3 3 6], shape=(3,), dtype=uint8)\n",
+      "Predictions:  [9 0 9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "test_data = pipeline.get_results(mode=\"test\")\n",
+    "test_data = network.transform(test_data, mode=\"test\")\n",
+    "print('Labels: ', test_data['y'])\n",
+    "print('Predictions: ', np.argmax(test_data['y_pred'].numpy(), axis=1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e781084b",
+   "metadata": {},
+   "source": [
+    "## Trace Debugging"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e11e319f",
+   "metadata": {},
+   "source": [
+    "### Conditional Debugging in Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cfb8c078",
+   "metadata": {},
+   "source": [
+    "What if our training is giving some weird result for the specific sample after the certain number of training epochs? How do you debug such sample data based on the specific conditions?\n",
+    "\n",
+    "In [Tutorial7](./t07_estimator.ipynb) we introduced the `Trace` and it's various use cases during the training. Since the trace allows us to control the training loop it is possible to add condition that suits your need to debug the training code."
```

Comment:
```suggestion
    "In [Tutorial7](./t07_estimator.ipynb) we introduced the `Trace` and it's various use cases during the training. Since the trace allows us to control the training loop it is possible to add conditions that suit your needs to debug the training code."
```